src/lib/layouts/workflow-history-layout.svelte

```diff
@@ -71,18 +88,73 @@
     {#if workflow?.callbacks?.length}
       <WorkflowCallbacks callbacks={workflow.callbacks} />
     {/if}
-    <div
-      class="flex flex-col items-center gap-2 py-2 xl:flex-row xl:justify-between xl:gap-8"
-    >
-      <h2>
-        {translate('workflows.event-history')}
-      </h2>
-    </div>
   </div>
 </div>
-<div class="px-2 pb-24 md:px-4 lg:px-8">
-  <div class="flex w-full flex-col border border-subtle">
-    <TimelineGraph {workflow} {groups} {workflowTaskFailedError} />
-    <EventSummary {groups} {history} />
+<div class="relative px-2 pb-24 md:px-4 lg:px-8">
+  <div
+    class="flex items-center justify-between gap-2 py-2 xl:gap-8"
+    class:sticky-header={!$minimizeEventView}
+  >
+    <h2>
+      {translate('workflows.event-history')}
+    </h2>
+    <div class="flex items-center gap-2">
+      <ToggleButtons>
+        {#if $eventViewType !== 'json'}
+          <ToggleButton
+            icon={reverseSort ? 'descending' : 'ascending'}
+            data-testid="zoom-in"
+            on:click={onSort}
+            >{reverseSort ? 'Descending' : 'Ascending'}</ToggleButton
+          >
+        {/if}
+        <ToggleButton
+          icon={$minimizeEventView ? 'minimize' : 'expand'}
+          data-testid="expandAll"
+          on:click={() => ($minimizeEventView = !$minimizeEventView)}
+          >{$minimizeEventView ? 'Minimized' : 'Expanded'}</ToggleButton
+        >
```

Comment:
If it's easy, it might be worth adding a tooltip here explaining what "minimize" and "expand" do.

Reply:
It's easy to add, now for what the words should be...